crates/solver/src/settlement.rs

```diff
@@ -456,6 +460,13 @@ impl Settlement {
         Ok(Self {
             encoder: merged,
             submitter: self.submitter,
+            score: match (self.score, other.score) {
+                (Some(left), Some(right)) => match (left, right) {
+                    (Score::Score(left), Score::Score(right)) => Some(Score::Score(left + right)),
+                    _ => None,
+                },
```

Comment:
Is equivalent and a little easier to read IMO:
```suggestion
                (Some(Score::Score(left)), Some(Score::Score(right))) => {
                    Some(Score::Score(left + right))
                }
```

Reply:
Even nicer.